app/models/post.rb

```diff
@@ -1,3 +1,15 @@
 class Post < ApplicationRecord
-  belongs_to :author
+  belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+
+  after_save :increment_author_posts_counter
+
+  def increment_author_posts_counter
+    author.increment!(:posts_counter)
+  end
```

Comment:
- [ OPTIONAL ] I recommend to make this method private here since you just need it here so it would be better to make it as a private method 🔥✅💯

Reply:
Alright, thanks for the great suggestion. I will definitely give it a trial.